File: app/src/main/java/com/example/ilya/myrocketapp/BaseAlgorithm.java
package com.example.ilya.myrocketapp;

import android.util.Pair;

import java.util.ArrayList;

public abstract class BaseAlgorithm {

    abstract ArrayList<Pair<Integer, Integer>> getResultOnUpdate();

    abstract void clearFields();

    abstract void run(int row, int column, boolean curColor);

    abstract boolean[][] getCurrentBitmapState();
}

File: app/src/main/java/com/example/ilya/myrocketapp/BaseAlgorithm.kt
package com.example.ilya.myrocketapp

import android.util.Pair

import java.util.ArrayList

abstract class BaseAlgorithm {

    internal abstract val resultOnUpdate: ArrayList<Pair<Int, Int>>

    internal abstract val currentBitmapState: Array<BooleanArray>

    internal abstract fun clearFields()

    internal abstract fun run(row: Int, column: Int, curColor: Boolean)
}
